move_base/src/move_base.cpp

```diff
@@ -958,6 +966,7 @@ namespace move_base {
         //check to see if we've reached our goal
         if(tc_->isGoalReached()){
           ROS_DEBUG_NAMED("move_base","Goal reached!");
+          publishFeedback(goal, 0.0, 0.0);
```

Comment:
Why do we need this? What if we finish 1m away with 1m tolerance?

Reply:
`publishFeedback(goal, 0.0, 0.0)` indicates to OMS, that we arrived to the goal in accuracy of tolerance.
OMS uses it for the correlation between feedback and make_plan.

Reply:
I don't understand this logic. OMS has perfect and only indication of mission completed or goals reached through mission shadow. Why does it need anything additional like dist_to_goal = 0?

Reply:
The logic is perfect, but  for intermediate waypoints  we don't have shadow.
We have to notify OMS, that robot has arrived.

Reply:
So, it's about intermediate goals only, right? Anyway, I think OMS is focused too much on tracking the intermediate points. What if you get dist_to_goal = 0.0 but the goal is not reached yet - like when doing final pivot rotation? Or when you get dist_to_goal = 1e-6? Checking for new goal seems more stable - if OMS gets new intermediate goal reported it means the previous one has been reached.

Reply:
Yes. It's about intermediate goals.I understand that dist_to_goal = 0.0 -> means that we near the goal, but not in the goal.
But for OMS 0.0  -> robot has arrived.
Let's leave the logic dist_to_goal = 0.0

Reply:
dist_to_goal = 0.0 doesn't mean we've arrived - robot can be 2 meters to the left or right, or trying to do the final pivot turning. We can leave it for now. But it's bad design which needs to be changed some day.

Reply:
Agree, let keep it for today.